plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/shipping/shipping-address.tsx

```diff
@@ -4,82 +4,87 @@
 import { __ } from '@wordpress/i18n';
 import { formatShippingAddress } from '@woocommerce/base-utils';
 import { useEditorContext } from '@woocommerce/base-context';
-import {
-	ShippingAddress as ShippingAddressType,
-	getSetting,
-} from '@woocommerce/settings';
+import { ShippingAddress as ShippingAddressType } from '@woocommerce/settings';
 import PickupLocation from '@woocommerce/base-components/cart-checkout/pickup-location';
 import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
 import { useSelect } from '@wordpress/data';
-
-/**
- * Internal dependencies
- */
-import ShippingLocation from '../../shipping-location';
-import { CalculatorButton, CalculatorButtonProps } from './calculator-button';
+import { useEffect, useState } from '@wordpress/element';
 
 export interface ShippingAddressProps {
-	showCalculator: boolean;
-	isShippingCalculatorOpen: boolean;
-	setIsShippingCalculatorOpen: CalculatorButtonProps[ 'setIsShippingCalculatorOpen' ];
+	setShippingCalculatorLabel: React.Dispatch<
+		React.SetStateAction< string >
+	>;
+	setShippingCalculatorAddress: React.Dispatch<
+		React.SetStateAction< string >
+	>;
 	shippingAddress: ShippingAddressType;
+	hasRates: boolean;
 }
 
-export type ActiveShippingZones = {
-	description: string;
-}[];
-
 export const ShippingAddress = ( {
-	showCalculator,
-	isShippingCalculatorOpen,
-	setIsShippingCalculatorOpen,
+	setShippingCalculatorLabel,
+	setShippingCalculatorAddress,
 	shippingAddress,
+	hasRates,
 }: ShippingAddressProps ): JSX.Element | null => {
 	const { isEditor } = useEditorContext();
 	const prefersCollection = useSelect( ( select ) =>
 		select( CHECKOUT_STORE_KEY ).prefersCollection()
 	);
-	const activeShippingZones: ActiveShippingZones = getSetting(
-		'activeShippingZones'
-	);
 
-	const hasMultipleAndDefaultZone =
-		activeShippingZones.length > 1 &&
-		activeShippingZones.some(
-			( zone: { description: string } ) =>
-				zone.description === 'Everywhere' ||
-				zone.description === 'Locations outside all other zones'
-		);
+	const [ showPickupLocation, setShowPickupLocation ] = useState( false );
+
+	useEffect( () => {
+		const hasFormattedAddress = !! formatShippingAddress( shippingAddress );
 
-	const hasFormattedAddress = !! formatShippingAddress( shippingAddress );
+		// If it's the editor and store has no formatted address,
+		// then the shipping calculator labels will not be modified.
+		// Shipping calculator will display the default label.
+		if ( isEditor && ! hasFormattedAddress ) {
```

Comment:
I would prefer to keep it. In the past, we have encountered cases where we were confused by the behavior of the shipping calculator. It might help us in the future to understand it easily.